pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -46,3 +76,183 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+// ScaleREST contains the REST storage for the Scale subresource of DeploymentConfigs.
+type ScaleREST struct {
+	registry     deployconfig.Registry
+	rcNamespacer kclient.ReplicationControllersNamespacer
+}
+
+// ScaleREST implements Patcher
+var _ = rest.Patcher(&ScaleREST{})
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &extensions.Scale{}
+}
+
+// Get retrieves (computes) the Scale subresource for the given DeploymentConfig name.
+func (r *ScaleREST) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	deploymentConfig, err := r.registry.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(directxman12): this is going to be a bit out of sync, since we are calculating it
+	// here and not as part of the deploymentconfig loop -- is there a better way of doing it?
+	totalReplicas, err := r.replicasForDeploymentConfig(deploymentConfig.Namespace, deploymentConfig.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	var specReplicas int
+
+	// current replicas reflects either the scale of the current deployment,
+	// or the scale of the RC template if no current deployment exists
+	controller, err := r.rcNamespacer.ReplicationControllers(deploymentConfig.Namespace).Get(util.LatestDeploymentNameForConfig(deploymentConfig))
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+
+		specReplicas = deploymentConfig.Template.ControllerTemplate.Replicas
+	} else {
+		specReplicas = controller.Spec.Replicas
+	}
+
+	return &extensions.Scale{
```

Comment:
It seems like this can built ahead of time, use the if blocks to set the fields you care about to make it a little more clear.

```go
// initialize return Scale
controller, err := big line
if err != nil{
    if !erro....{
    }

    ret.SpecReplicas = 
    ret.TotalReplicas = 
    return ret, nil
}

ret.specReplicas = 
ret.TotalReplicas = 
return ret, nil
```